src/net/sourceforge/kolmafia/request/FightRequest.java

```diff
@@ -7091,6 +7095,32 @@ private static void processP(Element node, TagStatus status) {
       FightRequest.logText(str, status);
     }
 
+    if (KoLCharacter.hasEquipped(ItemPool.TOY_CUPID_BOW)) {
+      if (Preferences.getInteger("cupidBowLastFamiliar")
+          == KoLCharacter.getEffectiveFamiliar().getId()) {
```

Comment:
Technically I don't think we want effective familiar here--that accounts for stuff like doppelshifter and chameleon. I guess chameleon can never equip a TCB, but doppelshifter can.